src/Resources/Coverage.php

```diff
@@ -5,9 +5,9 @@
 use Nesk\Rialto\Data\BasicResource;
 
 /**
- * @method void startJSCoverage(array<string, mixed> $options = null)
+ * @method void startJSCoverage(array $options = [])
  * @method mixed[] stopJSCoverage()
- * @method void startCSSCoverage(array<string, mixed> $options = null)
+ * @method void startCSSCoverage(array $options = [])
```

Comment:
Actually, those types are perfectly valid and I'm not the only one using them. This is supported by [PHPStan](https://phpstan.org/writing-php-code/phpdoc-types#general-arrays) and [Psalm](https://psalm.dev/docs/annotating_code/type_syntax/array_types/#generic-arrays).

Also, [PHPStorm should support those syntaxes](https://blog.jetbrains.com/phpstorm/2020/07/phpstan-and-psalm-support-coming-to-phpstorm/), maybe you didn't update to the latest version?

Reply:
![Снимок экрана 2021-02-09 в 22 20 07](https://user-images.githubusercontent.com/34161928/107415957-f8acc900-6b24-11eb-8221-253ce58100b7.png)

Im using latest available version. In your link such types support implemented as plugins. Probably something went wrong with devs plans and they were not added/enabled by default...

Also there can be problems with other IDEs, since its not standart php syntax. 

Maybe we can use php 8 annotations? But i did not used them before.

Reply:
I just checked release notes https://blog.jetbrains.com/phpstorm/2020/12/phpstorm-2020-3-release/#psalm_and_phpstan_support 

Such types can be used in project if Psalm or PHPStan added in composer and inspections enabled in settings. 

Also custom attributes added to phpStorm, but i doubt they will not be supported in different IDE's https://blog.jetbrains.com/phpstorm/2020/12/phpstorm-2020-3-release/#custom_php_attributes_in_phpstorm . So its not best idea to use #[ArrayShape].

I will try to add psalm tom composer and setup my IDE. Maybe most of my edits will be not needed after that.

Reply:
Not all IDE's may support these annotations today, but it's really helping if you properly configure your CI with a static analysis tool like Psalm, PHPStan, etc. And support will improve in the next months/years. 🙂

For example, recently I was using [getsentry/sentry-php](https://github.com/getsentry/sentry-php) package, and their extended annotations saved me multiple times.

Reply:
I checked phpStan and Psalm support in phpStorm, and they both don't support such syntax in `@method` block
![Снимок экрана 2021-02-16 в 18 08 17](https://user-images.githubusercontent.com/34161928/108081374-f3e69880-7081-11eb-8105-cac84a782903.png)

But same syntax in `@param` or `@return` blocks works ok:
![Снимок экрана 2021-02-16 в 18 09 52](https://user-images.githubusercontent.com/34161928/108081583-2d1f0880-7082-11eb-9cc1-d59131d17b61.png)

So my plan is: 

1. add phpStan to composer dev, to enable as much phpDocs syntax in IDE as possible
2. Revert changes in traits
3. Fix bugs with JsHandle/JsFunction
4. Fix missing methods in Puppeteer class (maybe exclude it from docs generation?)

What do you think? Do you insist on full namespaces instead of FQCN?